src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,105 @@
+                try {
+                    int index = Integer.parseInt(splitString[1]);
+                    if (instr.equals("unmark")) {
+                        taskList.unmarkTask(index);
+                    } else if (instr.equals("mark")) {
+                        taskList.markTask(index);
+                    } else if (instr.equals("delete")) {
+                        taskList.deleteTask(index);
+                    } else {
+                        throw new DukeException("You have entered an invalid instruction");
+                    }
+                } catch (DukeException e) {
+                    System.out.println(e);
+                }
+            }
+        } else if (input.contains("find")) { //input is find
+            String[] splitString = input.split(" ", 2);
+            if (splitString.length < 2) {
+                throw new DukeException("Please input the keyword(s) for find");
+            }
+            String command = splitString[0];
+            String text = splitString[1];
+            if (command.equals("find")) {
+                taskList.find(text);
+            } else {
+                throw new DukeException("You have entered an invalid command");
+            }
+        } else if (input.contains("todo") || input.contains("event") || input.contains("deadline")) {
+            //input is a new type of task
+            //identify type of task
+            String[] stringArray = input.split(" ", 2);
+
+            //task has no task detail/name
+            if (stringArray.length < 2) {
+                throw new DukeException("Description of task cannot be empty!");
+            }
+
+            String taskType = stringArray[0];
+            String taskDetails = stringArray[1];
+
+            Task newTask = new Task("");
+
+            if (taskType.equals("todo")) {
+                newTask = new Todo(taskDetails);
+            } else if (taskType.equals("deadline")) {
+                String[] stringSplit = taskDetails.split("/by");
+                if (stringSplit.length < 2) {
+                    throw new DukeException("Description of deadline must include a date/time! Did you miss out a /by?");
+                }
+                String details = stringSplit[0].trim();
+                String dateTime = stringSplit[1].trim();
+                newTask = new Deadline(details,dateTime);
+            } else if (taskType.equals("event")) {
+                String[] splitString = taskDetails.split("/at");
+                if (splitString.length < 2) {
+                    throw new DukeException("Description of event must include a date/time! Did you miss out a /at?");
+                }
+                String details = splitString[0].trim();
+                String dateTime = splitString[1].trim();
+                newTask = new Event(details,dateTime);
+            }
+
+            taskList.addTask(newTask);
+        } else {
+            throw new DukeException("no such task type");
+        }
```

Comment:
Should the individual code sections of the different commands be extracted? for example 'handleTodo(String input)' and 'handleDelete(String input)'?